python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -436,6 +442,24 @@ def _executeQuantaInProcess(self, graph, butler):
                 self.quantumExecutor.execute(qnode.taskDef, qnode.quantum, butler)
                 successCount += 1
             except Exception as exc:
+                if self.pdb and self.numProc == 1:
```

Comment:
Check for `numProc` is probably not needed here, this code always runs in-process which means no parallelism.
I'd probably add a paranoid check that stdin/stdout are connected to a terminal (`sys.std{in,out}.isatty()`).

Reply:
I like the tty check. That also catches the case where we're running with multiple processes, which was my main concern.